sunpy/instr/goes.py

```diff
@@ -147,28 +162,32 @@ def temp_em(goeslc, abundances="coronal", download=False, download_dir=DATA_PATH
     2014-01-01 00:00:06  7e-07  7e-06  11.28295376  4.78577516e+48
 
     """
-
     # Check that input argument is of correct type
-    if not isinstance(goeslc, lightcurve.GOESLightCurve):
-        raise TypeError("goeslc must be a GOESLightCurve object.")
+    if not isinstance(goeslc, lightcurve.LightCurve):
+        raise TypeError("goeslc must be a LightCurve object.")
 
-    # Find temperature and emission measure with goes_chianti_tem
-    temp, em = goes_chianti_tem(goeslc.data.xrsb, goeslc.data.xrsa,
-                                satellite=goeslc.meta["TELESCOP"].split()[1],
-                                date=goeslc.data.index[0],
-                                abundances=abundances, download=download,
-                                download_dir=download_dir)
+    # Find temperature and emission measure with _goes_chianti_tem
+    temp, em = _goes_chianti_tem(Quantity(goeslc.data.xrsb, unit=u.W/(u.m)**2),
```

Comment:
You should explicitly mention in the docs you expect the LC data to be in these units. Due to the lack of unit support in LightCurve, this is not guaranteed.